homeassistant/components/solax/sensor.py

```diff
@@ -28,11 +29,13 @@
 from homeassistant.helpers.event import async_track_time_interval
 
 DEFAULT_PORT = 80
+DEFAULT_PASSWORD = ''
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_IP_ADDRESS): cv.string,
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
```

Comment:
We no longer allow integrations to add or change a platform YAML configuration.

More information on this can be found in Architecture Decision Record:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>

Please note that this integration connects to a device or service and another Architecture Decision Record applies that disallows the use of YAML configuration in favor of a configuration flow via the UI:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

As these changes often involve a bit of work and some significant shift in the current code, we will close this PR for now.

We (and our community!) would really appreciate it if you could start on the refactoring of this integration in a new PR.

Thanks already! :+1:

Reply:
Hello. Thank you for the feedback.

I'm fairly new to HA development, but I will take the time to refactor this integration to a UI flow.

Reply:
Thanks @tokenize47 🎉 

If you need help or questions, feel free to reach out to us on the Home Assistant Discord chat server.